magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,417 @@
+    dHc = 1320 * u.m
+
+    data = []
+    dl1_params = read_table(input_file, "/events/parameters")
+    dl1_images = read_table(input_file, "/events/dl1/image_" + str(telid))
+
+    inds = np.where(
+        np.logical_and(dl1_params["intensity"] > 0.0, dl1_params["tel_id"] == telid)
+    )[0]
+    for index in inds:
+        event_id_lst = dl1_params["event_id_lst"][index]
+        obs_id_lst = dl1_params["obs_id_lst"][index]
+        event_id = dl1_params["event_id_magic"][index]
+        obs_id = dl1_params["obs_id_magic"][index]
+        if telid == 1:
+            event_id, obs_id = event_id_lst, obs_id_lst
+
+        pointing_az = dl1_params["pointing_az"][index]
+        pointing_alt = dl1_params["pointing_alt"][index]
+        pointing_alt_deg = np.rad2deg(pointing_alt)
+        zenith = 90.0 - pointing_alt_deg
+
+        x = dl1_params["x"][index] * u.m
+        y = dl1_params["y"][index] * u.m
+        psi = dl1_params["psi"][index] * u.deg
+        wl = dl1_params["width"][index] / dl1_params["length"][index]
+        log_intensity = np.log10(dl1_params["intensity"])[index]
+        timestamp = dl1_params["timestamp"][index]
+        time_diff = dl1_params["time_diff"][index]
+        n_islands = dl1_params["n_islands"][index]
+        n_pixels = dl1_params["n_pixels"][index]
+        signal_pixels = dl1_params["n_pixels"][index]
+
+        trans = 0.58  # transmission of the cloud
+        trans = trans ** (1 / np.cos(zenith))  # .to_value("")
+        nlayers = 10  # to simplify calculations with the model cloud is considered to be composed of multiple layers
+        Hcl = np.linspace(Hc, Hc + dHc, nlayers)  # position of each layer
+        transl = trans_height(Hcl, Hc, dHc, trans)  # transmissions of each layer
+        transl = np.append(
+            transl, transl[-1]
+        )  # add one more element for heights above the top of the cloud
+
+        alt_rad = np.deg2rad(dl1_params["alt"][index])
+        az_rad = np.deg2rad(dl1_params["az"][index])
+
+        eff_focal = focal_eq[telid]  # *u.m # m
+        m2deg = np.rad2deg(1) / eff_focal * u.degree
+
+        impact = dl1_params["impact"][index] * u.m
+        psi = dl1_params["psi"][index] * u.deg
+        cog_x = (dl1_params["x"][index] * m2deg).value * u.deg
+        cog_y = (dl1_params["y"][index] * m2deg).value * u.deg
+
+        # Source position
+        pointing_altaz = SkyCoord(alt=alt_rad * u.rad, az=az_rad * u.rad, frame=AltAz())
+        telescope_pointing = SkyCoord(
+            alt=dl1_params["pointing_alt"][index] * u.rad,
+            az=dl1_params["pointing_az"][index] * u.rad,
+            frame=AltAz(),
+        )
+
+        tel_frame = TelescopeFrame(telescope_pointing=telescope_pointing)
+        tel = pointing_altaz.transform_to(tel_frame)
+
+        src_x = tel.fov_lat.to(u.deg).value
+        src_y = tel.fov_lon.to(u.deg).value
+
+        # Transform to Engineering camera
+        src_x, src_y = -src_y * u.deg, -src_x * u.deg
```

Comment:
you just dropped deg and here you multiply by it again

Reply:
Changed